src/vcModals.cpp

```diff
@@ -656,10 +656,89 @@ void vcModals_DrawProjectInfo(vcState *pProgramState)
     if (ImGui::BeginChild("##infoModal", ImVec2(-1, -30), true))
     {
       if (udProjectNode_GetMetadataString(pProgramState->activeProject.pRoot, "information", &pInfo, "") == udE_Success)
-        ImGui::TextWrapped("%s", pInfo);
+      {
+        if (udStrcmp(pProgramState->projectInfoTextures.pLastInfoText, pInfo) != 0)
+        {
+          pProgramState->projectInfoTextures.pLastInfoText = udStrdup(pInfo);
+          for (vcTexture *pTexture : pProgramState->projectInfoTextures.textures)
+            vcTexture_Destroy(&pTexture);
+          for (const char *pStr : pProgramState->projectInfoTextures.infoStrings)
+            udFree(pStr);
+          pProgramState->projectInfoTextures.infoStrings.Clear();
+          pProgramState->projectInfoTextures.textures.Clear();
+          pProgramState->projectInfoTextures.textureSizes.Clear();
+          
+          // Discover Images
+          int firstIndex = 0;
+          int startStrIndex = 0;
+          while (true)
+          {
+            size_t startImagePos = 0;
+            size_t openBracketPos = 0;
+            size_t closeBracketPos = 0;
+            size_t openParanthesisPos = 0;
+            size_t closeParanthesisPos = 0;
+
+            udStrchr(pInfo + firstIndex, "!", &startImagePos);
+            startImagePos += firstIndex;
+            udStrchr(pInfo + startImagePos, "[", &openBracketPos);
+            openBracketPos += startImagePos;
+            udStrchr(pInfo + openBracketPos, "]", &closeBracketPos);
+            closeBracketPos += openBracketPos;
+            udStrchr(pInfo + closeBracketPos, "(", &openParanthesisPos);
+            openParanthesisPos += closeBracketPos;
+            udStrchr(pInfo + openParanthesisPos, ")", &closeParanthesisPos);
+            closeParanthesisPos += openParanthesisPos;
+            
+            const int64_t invalidLen = udStrlen(pInfo);
+
+            if (startImagePos == invalidLen ||
+              openBracketPos == invalidLen ||
+              closeBracketPos == invalidLen ||
+              openParanthesisPos == invalidLen ||
+              closeParanthesisPos == invalidLen)
+              break;
+
+            size_t lastCharPos = closeParanthesisPos;
+            if (openBracketPos == startImagePos + 1 &&
+              closeBracketPos > openBracketPos &&
+              openParanthesisPos == closeBracketPos + 1 &&
+              closeParanthesisPos > openParanthesisPos)
```

Comment:
I'm not sure that `openBracketPos` is necessary, it should always be `startImagePos + 1`, so instead of using `udStrchr` to find it, just check that the character after `startImagePos` is `[` instead. Same goes for `openParenthesisPos`. Additionally, `closeBracketPos` is calculated from `pInfo + openBracketPos`, so `closeBracketPos > openBracketPos` will always be true by this point. Same goes for `closeParenthesisPos > openParenthesisPos`.

Reply:
I need the position for splitting up the string so i think it's necessary. this function doesn't get called as much now, so the performance hit i think it acceptable

Reply:
> I need the position for splitting up the string so i think it's necessary

You don't, you only need to know where `!`, `]` and `)` are, the characters that follow `!` and `]` are `[` and `(`, respectively. You can remove `openBracketPos` and `openParenthesisPos` and use `startImagePos + 1` and `closeBracketPos + 1`, respectively. Instead of checking `openBracketPos == startImagePos + 1` check `pInfo[startImagePos + 1] == '['`, similarly for `openParenthesis`.

> this function doesn't get called as much now, so the performance hit i think it acceptable

If this was about performance, I would be explaining how you should be doing this without allocating so much memory, a single `udStrdup` is sufficient. This is about readability and making your intentions clear.